tests/hla/base_classes.py

```diff
@@ -0,0 +1,178 @@
+
+from drizzlepac.hlautils.astroquery_utils import retrieve_observation
+
+# Base classes for actual tests.
+# NOTE: Named in a way so pytest will not pick them up here.
+class BaseTest(object):
+    prevdir = os.getcwd()
+    use_ftp_crds = True
+    timeout = 30  # seconds
+    tree = 'dev'
+    docopy = False
+
+    # Numpy default for allclose comparison
+    rtol = 1e-6
+    atol = 0
+
+    # To be defined by instrument/test
+    input_repo = ''  # e.g., 'drizzlepac' or 'jwst-pipeline'
+    results_root = None  # e.g., 'drizzlepac-results'
+    input_loc = ''  # root directory for 'input' files
+    ref_loc = []    # root path for 'truth' files: ['test1','truth'] or ['test3']
+    ignore_keywords = []
+    ignore_table_keywords = []
+    ignore_fields = []
+    ignore_hdus = []
+
+    # To be defined by individual test
+    subdir = ''
+    curdir = ''
+
+    @pytest.fixture(autouse=True)
+    def setup_class(self, tmpdir, envopt):
+        """
+        Run test in own dir so we can keep results separate from
+        other tests.
+        """
+        # create working directory specified for the test
+        if not tmpdir.ensure(self.subdir, dir=True):
+            curdir = tmpdir.mkdir(self.subdir).strpath
+        else:
+            curdir = tmpdir.join(self.subdir).strpath
+        os.chdir(curdir)
+        self.curdir = curdir
+
+        # This controls astropy.io.fits timeout
+        conf.remote_timeout = self.timeout
+
+        # Update tree to point to correct environment
+        self.tree = envopt
+
+        # Configure environment for tests
+        self.set_environ()
+
+    def teardown_class(self):
+        """Reset path and variables."""
+        conf.reset('remote_timeout')
+        os.chdir(self.prevdir)
+        if self.use_ftp_crds and self.prevref is not None:
+            os.environ[self.refstr] = self.prevref
+
+    def set_environ(self):
+        """Set environment variables for test environment to be used
+
+           Each class/test will define what needs to be set, if anything.
+           For example, control use of Astrometry updates for HST with:
+               os.environ['ASTROMETRY_STEP_CONTROL'] = 'OFF'
+        """
+        pass
+
+    def get_input_path(self):
+        """
+        Return path within repository to remote source of input data
+        """
+        path = [self.input_repo, self.tree, self.input_loc]
+
+        return path
+
+    def get_data(self, *args, **kwargs):
+        """
+        Download `filename` into working directory using
+        `artifactory_helpers/get_bigdata()` or 
+        `astroquery_utils.retrieve_observation`.  Use of `astroquery_utils`
+	will allow getting data directly from MAST via astroquery.
+
+        Returns
+        --------
+        local_files : list
+            This will return a list of all the files downloaded with 
+            the full path to the local copy of the file.
+        """
+        if len(args[0]) == 9: # Only a rootname provided
```

Comment:
If that length ever changes...

Reply:
Understood, but in this case I am willing to risk it.